src/theory/sets/theory_sets.cpp

```diff
@@ -201,37 +201,6 @@ bool TheorySets::isEntailed( Node n, bool pol ) {
 
 /**************************** eq::NotifyClass *****************************/
 
-bool TheorySets::NotifyClass::eqNotifyTriggerPredicate(TNode predicate,
-                                                       bool value)
-{
-  Debug("sets-eq") << "[sets-eq] eqNotifyTriggerPredicate: predicate = "
-                   << predicate << " value = " << value << std::endl;
-  if (value)
-  {
-    return d_theory.propagate(predicate);
-  }
-  return d_theory.propagate(predicate.notNode());
-}
-
-bool TheorySets::NotifyClass::eqNotifyTriggerTermEquality(TheoryId tag,
-                                                          TNode t1,
-                                                          TNode t2,
-                                                          bool value)
-{
-  Debug("sets-eq") << "[sets-eq] eqNotifyTriggerTermEquality: tag = " << tag
-                   << " t1 = " << t1 << "  t2 = " << t2 << "  value = " << value
-                   << std::endl;
-  d_theory.propagate(value ? t1.eqNode(t2) : t1.eqNode(t2).negate());
-  return true;
```

Comment:
Just to satisfy my heart, it seems this function always returns true, however, the new one `TheoryEqNotifyClass::eqNotifyTriggerTermEquality` returns false when there is a conflict. 
Was it the case that sets solver never reaches this line in a state of a conflict? or the code was not correct, and you actually fixed a hidden bug :)

Reply:
Yes, I'm surprised the code was initially like this.

I believe this does fix a hidden bug, albeit probably a minor one.  The return value is simply for telling some of the internals in the sets solver (e.g. in its equality engine) that it should terminate because we are going to backtrack.